httpcache.go

```diff
@@ -28,6 +28,16 @@ const (
 	XFromCache = "X-From-Cache"
 )
 
+var cacheableResponseCodes map[int]bool = map[int]bool{
+	http.StatusOK:                    true, // 200
+	http.StatusNonAuthoritativeInfo:  true, // 203
+	http.StatusMultipleChoices:       true, // 300
+	http.StatusMovedPermanently:      true, // 301
+	http.StatusFound:                 true, // 302
+	http.StatusNotFound:              true, // 404
+	http.StatusGone:                  true, // 410
+}
```

Comment:
Just presence of the status in the map should be sufficient information, so consider the following alternative:

```Go
var cacheableResponseCodes = map[int]struct{}{
    http.StatusOK:                    {}, // 200
    http.StatusNonAuthoritativeInfo:  {}, // 203
    ...
}
```

And checked like this:

```Go
if _, cacheable := cacheableResponseCodes[code]; !cacheable {
    return false
}
```